torch/_dynamo/variables/builder.py

```diff
@@ -1187,6 +1185,15 @@ def wrap_to_fake_tensor_and_record(
                 # NB: both static and dynamic have precedence over
                 automatic_dynamic = curr_sizes is None or curr_sizes[i] is None
 
+                # Reflect the user directive in the frame_state
+                # For dynamic, apply None always
+                if marked_dynamic:
+                    curr_sizes[i] = None
+
+                # For static - undo setting None
+                if marked_static:
+                    curr_sizes[i] = e.size()[i]
```

Comment:
No, I don't think you can implement it this way. I'd suggest splitting this into a separate PR.

With mark static, this won't make it sticky. The idea behind mark static is to force us to generate new kernels for each size, even if the size changes. If you only mark static by clearing the frame state, if we toggle this frame a few more times the frame state will go back to None (dynamic).

That being said, I'm not convinced static should be sticky. Instead we probably just need some guard thing that ensures that dynamic kernels reject incoming tensors that are marked static so that we recompile.

Reply:
I will split, but your points make good sense. I won't do the static thing in the split.